js/state/siteUtil.js

```diff
@@ -600,14 +600,13 @@ module.exports.isImportedBookmark = function (siteDetail) {
  */
 module.exports.isHistoryEntry = function (siteDetail) {
   if (siteDetail && typeof siteDetail.get('location') === 'string') {
-    if (siteDetail.get('location').startsWith('about:')) {
+    const tags = siteDetail.get('tags')
+    if (siteDetail.get('location').startsWith('about:') ||
+      module.exports.isDefaultEntry(siteDetail) ||
+      isBookmarkFolder(tags)) {
       return false
     }
-    if (module.exports.isDefaultEntry(siteDetail)) {
-      // This is a Brave default newtab site
-      return false
-    }
-    return !!siteDetail.get('lastAccessedTime') && !isBookmarkFolder(siteDetail.get('tags'))
+    return !!siteDetail.get('lastAccessedTime') || !tags || tags.size === 0
```

Comment:
@ayumi actually the crash was caused by a history entry which did not have `lastAccessedTime`, so changing it to `!!siteDetail.get('lastAccessedTime') && (!tags || tags.size === 0)` would cause it to crash again. 

when i sync to `forsaken...`, the synced bookmarks appear in `about:history`. this seems like the correct behavior to me because every bookmark is by definition also a history entry.